zhusuan/hmc.py

```diff
@@ -460,6 +460,11 @@ def iss():
                     self.q, p, current_q, current_p,
                     get_log_posterior, mass, self.data_axes)
 
+            old_log_prob = tf.check_numerics(old_log_prob, 'HMC: old_log_prob is NaN! Try better initialization?')
+            is_nan = tf.logical_or(tf.is_nan(acceptance_rate),
+                                    tf.is_nan(new_log_prob))
+            acceptance_rate = tf.where(is_nan,
+                    tf.zeros_like(acceptance_rate), acceptance_rate)
```

Comment:
If all nan's get rejected, the old_log_prob will only be nan just after initialization. Am I right?